Apps/W1/EDocumentConnectors/Continia/app/src/Implementation/ContiniaEDocumentProcessing.Codeunit.al

```diff
@@ -0,0 +1,281 @@
+    var
+        ApiRequests: Codeunit "Continia Api Requests";
+        XMLFileToken: Text;
+        XmlDocNode: XmlNode;
+        XMLFileTokenNode: XmlNode;
+        XMLDocumentIdNode: XmlNode;
+        InStream: InStream;
+        DocumentInfoXml: XmlDocument;
+        CurrentEDocumentNode: XmlNode;
+    begin
+        DocumentMetadata.CreateInStream(InStream, TextEncoding::UTF8);
+        XmlDocument.ReadFrom(InStream, DocumentInfoXml);
+        DocumentInfoXml.SelectSingleNode('document', CurrentEDocumentNode);
+
+        CurrentEDocumentNode.SelectSingleNode('document_id', XMLDocumentIdNode);
+        Evaluate(EDocument."Continia Document Id", XMLDocumentIdNode.AsXmlElement().InnerText);
+        EDocument.Modify();
+
+        CurrentEDocumentNode.SelectSingleNode('xml_document', XmlDocNode);
+        XmlDocNode.SelectSingleNode('file_token', XMLFileTokenNode);
+        XMLFileToken := XMLFileTokenNode.AsXmlElement().InnerText;
+
+        // Download XML file from XMLFileToken and save to TempBlob
+        ApiRequests.DownloadFileFromUrl(XMLFileToken, ReceiveContext);
+    end;
+
+    internal procedure MarkFetched(var EDocument: Record "E-Document"; var EDocumentService: Record "E-Document Service"; var DocumentBlob: Codeunit "Temp Blob"; ReceiveContext: Codeunit ReceiveContext)
+    var
+        ApiRequests: Codeunit "Continia Api Requests";
+    begin
+        // Mark document as processed in Continia Online
+        ApiRequests.MarkDocumentAsProcessed( EDocument."Continia Document Id", ReceiveContext);
+    end;
+
+    procedure GetCancellationStatus(var EDocument: Record "E-Document"; var EDocumentService: Record "E-Document Service"; ActionContext: Codeunit ActionContext) Success: Boolean
+    var
+        ApiRequests: Codeunit "Continia Api Requests";
+    begin
+        exit(ApiRequests.CancelDocument(EDocument."Continia Document Id", ActionContext));
+    end;
+
+    [EventSubscriber(ObjectType::Page, Page::"E-Document Service", OnAfterValidateEvent, "Service Integration V2", true, true)]
+    local procedure OnAfterValidateServiceIntegration(var Rec: Record "E-Document Service")
+    begin
+        if Rec."Service Integration V2" <> Rec."Service Integration V2"::Continia then
+            exit;
+
+        if Rec."Document Format" = Rec."Document Format"::"Data Exchange" then
+            Error(DocumentFormatUnsupportedErr);
+    end;
+
+    [EventSubscriber(ObjectType::Page, Page::"E-Document Service", OnAfterValidateEvent, "Export Format", true, true)]
+    local procedure OnAfterValidateDocumentFormat(var Rec: Record "E-Document Service")
+    begin
+        if Rec."Service Integration V2" <> Rec."Service Integration V2"::Continia then
+            exit;
+
+        if Rec."Document Format" = Rec."Document Format"::"Data Exchange" then
+            Error(DocumentFormatUnsupportedErr);
```

Comment:
There could be more formats in other local countries. So maybe better to check against positive list.